lib/active_storage_validations/matchers/limit_validator_matcher.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+require 'pry'
+require_relative 'concerns/active_storageable.rb'
+require_relative 'concerns/allow_blankable.rb'
+require_relative 'concerns/attachable'
+require_relative 'concerns/contextable.rb'
+require_relative 'concerns/messageable.rb'
+require_relative 'concerns/rspecable.rb'
+require_relative 'concerns/validatable.rb'
+
+module ActiveStorageValidations
+  module Matchers
+    def validate_limits_of(name)
+      LimitValidatorMatcher.new(name)
+    end
+
+    class LimitValidatorMatcher
+      include ActiveStorageable
+      include AllowBlankable
+      include Attachable
+      include Contextable
+      include Messageable
+      include Rspecable
+      include Validatable
+
+      def initialize(attribute_name)
+        initialize_allow_blankable
+        initialize_contextable
+        initialize_messageable
+        initialize_rspecable
+        @attribute_name = attribute_name
+      end
+
+      def description
+        "validate the limit files of :#{@attribute_name}"
+      end
+
+      def failure_message
+        message = ["is expected to validate limit file of :#{@attribute_name}"]
+        build_failure_message(message)
+        message.join("\n")
+      end
+
+      def min(number)
+        @min = number
+        self
+      end
+
+      def min_between(range)
+        @min, @max = range.first, range.last
+        self
+      end
+
+      def max(number)
+        @max = number
+        self
+      end
+
+      def max_between(range)
+        @min, @max = range.first, range.last
+        self
+      end
+
+      def matches?(subject)
+        @subject = subject.is_a?(Class) ? subject.new : subject
+        # binding.pry
+        is_a_valid_active_storage_attribute? &&
+          is_context_valid? &&
+          is_custom_message_valid? &&
+          file_number_not_smaller_than_min? &&
+          file_number_equal_min? &&
+          file_number_larger_than_min? &&
+          file_number_smaller_than_max? &&
+          file_number_equal_max? &&
+          file_number_not_larger_than_max?
+      end
+
+      private
+
+      def build_failure_message(message)
+        return unless @failure_message_artefacts.present?
+
+        message << "  but there seem to have issues with the matcher methods you used, since:"
+        @failure_message_artefacts.each do |error_case|
+          message << "  validation failed when provided with a #{error_case[:count]} limit file"
```

Comment:
reading the failure_message, it could rather be: `" validation failed when provided with a #{error_case[:count]} file(s)"`